### PR TITLE
devicestate: add support for cloud.cfg.d config from the gadget

### PR DESCRIPTION
**Pull request by mvo5**
This commit adds support for cloud init data from the gadget. If
the gadget snap has a dir `cloud.cfg.d` all the content of that
dir is copied to the `ubuntu-data` partition during the install
phase.

### CODE REVIEW

File: overlord/devicestate/handlers_install.go [hunk 1]
```diff
@@ -43,6 +43,28 @@ var (
 	sysconfigConfigureRunSystem = sysconfig.ConfigureRunSystem
 )
 
+func setSysconfigCloudOptions(opts *sysconfig.Options, gadgetDir string, model *asserts.Model) {
+	// 1. check cloud.cfg.d in the gadget snap, this is always ok regardless
+	//    of grade
+	cloudCfg := filepath.Join(gadgetDir, "cloud.cfg.d")
+	if osutil.IsDirectory(cloudCfg) {
+		opts.CloudInitSrcDir = cloudCfg
```
Comment (bboozzoo) [inline overlord/devicestate/handlers_install.go:51]: What if both locations are present?

Comment (mvo5) [reply overlord/devicestate/handlers_install.go:51]: A good question, right now the gadget wins but I guess that is a matter for debate and we could allow more customization for dangerous models.

Comment (mvo5) [reply overlord/devicestate/handlers_install.go:51]: Maybe in a followup so that cloud people can prepare images now?

Comment (bboozzoo) [reply overlord/devicestate/handlers_install.go:51]: Yeah, a follow up is fine. Let's unblock the cloud guys first.

Comment (bboozzoo) [reply overlord/devicestate/handlers_install.go:51]: Hm maybe we should leave a TODO at least.

**Pull request merged**